pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -204,7 +206,7 @@ func (r *ReconcileKnativeServing) install(instance *servingv1alpha1.KnativeServi
 		}
 	}
 	if err != nil {
-		instance.Status.MarkInstallFailed(err.Error())
+		instance.Status.MarkInstallFailed("Install failed with message", err.Error())
```

Comment:
The disambiguation logic should be added here I think. Otherwise we'll update status twice and have a weird flickering from "Install failed with message" to "Install in progress"

Reply:
Do you mean we need to check error type here and update status accordingly ?

Reply:
yep. Instead of above. `updateStatus` is called here automatically too.